### PR TITLE
Add IPv4/6 private address space checkers

### PR DESCRIPTION
**Pull request by jukkar**
Add helper functions that allow user to check whether the given IPv4/6 address is from private address space.

### CODE REVIEW

File: include/zephyr/net/net_ip.h [hunk 1]
```diff
@@ -736,6 +736,34 @@ static inline bool net_ipv4_is_ll_addr(const struct in_addr *addr)
...
+{
+	uint32_t masked_24, masked_16, masked_12, masked_10, masked_8;
+
+	masked_24 = ntohl(UNALIGNED_GET(&addr->s_addr)) & 0xFFFFFF00;
+	masked_16 = masked_24 & 0xFFFF0000;
+	masked_12 = masked_24 & 0xFFF00000;
+	masked_10 = masked_24 & 0xFFC00000;
+	masked_8 = masked_24 & 0xFF000000;
+
+	return   masked_8 == 0x0A000000 || /* 10.0.0.0/8 */
```
Comment (rlubos) [inline include/zephyr/net/net_ip.h:758]: Triple space after `return`.

File: include/zephyr/net/net_ip.h [hunk 2]
```diff
@@ -736,6 +736,34 @@ static inline bool net_ipv4_is_ll_addr(const struct in_addr *addr)
...
+		masked_10 == 0x64400000 || /* 100.64.0.0/10 */
+		masked_12 == 0xAC100000 || /* 172.16.0.0/10 */
```
Comment (rlubos) [inline include/zephyr/net/net_ip.h:760]: Netmask in the comment is wrong (should be 12).

File: include/zephyr/net/net_ip.h [hunk 3]
```diff
@@ -878,6 +906,26 @@ static inline bool net_ipv6_is_global_addr(const struct in6_addr *addr)
...
+ * @return True if it is, false otherwise.
+ */
+static inline bool net_ipv6_is_private_addr(const struct in6_addr *addr)
+{
+	uint32_t masked_32, masked_7;
+
+	masked_32 = ntohl(UNALIGNED_GET(&addr->s6_addr32[0]));
+	masked_7 = masked_32 & 0xfc000000;
+
+	return  masked_32 == 0x20010db8 || /* 2001:db8::/32 */
```
Comment (rlubos) [inline include/zephyr/net/net_ip.h:925]: Double space after `return`.

File: tests/net/ip-addr/src/main.c [hunk 4]
```diff
@@ -602,6 +602,104 @@ ZTEST(ip_addr_fn, test_ipv6_mesh_addresses)
...
+ZTEST(ip_addr_fn, test_private_ipv6_addresses)
+{
+	bool ret;
+	struct {
+		struct in6_addr addr;
+		bool is_private;
+	} addrs[] = {
+		{
+			.addr = { { { 0x20, 0x01, 0x0d, 0xb8, 0, 0, 0, 0,
+						0, 0, 0, 0, 0, 0, 0x99, 0x1 } } },
```
Comment (rlubos) [inline tests/net/ip-addr/src/main.c:614]: This indentation looks a bit weird, I don't think it was intended? Same in others below.

**Review by rlubos — COMMENTED** [review]
Looks good, just some nits.

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**